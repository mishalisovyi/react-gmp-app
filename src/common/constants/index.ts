export const Icon = {
  VERTICAL_ELLIPSIS: String.fromCharCode(8942),
  MULTIPLICATION_X: String.fromCharCode(10005),
  TELEPHONE_RECORDER: String.fromCharCode(8981),
};

export const API_URL = 'http://localhost:4000';

export const HTTP_RESPONSE_STATUS_NO_CONTENT = 204;
