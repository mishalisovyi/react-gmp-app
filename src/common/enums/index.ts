export { ButtonType } from './button-type.enum';
export { HttpMethod } from './http-method.enum';
