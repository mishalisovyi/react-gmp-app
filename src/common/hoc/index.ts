export { withPortalRendering } from './withPortalRendering/withPortalRendering';
export { ModalWindow } from './ModalWindow/ModalWindow';
