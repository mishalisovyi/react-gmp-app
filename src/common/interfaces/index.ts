export { EnumerableComponentProps } from './EnumerableComponentProps';
export { ModalWindowProps, ModalWindowActionButtonProps } from './ModalWindowProps';
export { SelectListData, SelectListValue } from './SelectListData';
