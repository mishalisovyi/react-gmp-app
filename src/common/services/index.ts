export {
  sendGetRequest,
  sendPostRequest,
  sendPutRequest,
  sendDeleteRequest,
} from './http.service';
