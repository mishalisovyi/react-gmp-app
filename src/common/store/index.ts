export * from './actions';
export * from './models';
export { commonReducer } from './reducer';
