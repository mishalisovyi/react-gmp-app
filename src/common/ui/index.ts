export { Button } from './Button/Button';
export { Divider } from './Divider/Divider';
export { DropdownMenu } from './DropdownMenu/DropdownMenu';
export { FilterSelectList } from './FilterSelectList/FilterSelectList';
export { LabelPrimary } from './LabelPrimary/LabelPrimary';
export { LabelSecondary } from './LabelSecondary/LabelSecondary';
export { Logo } from './Logo/Logo';
export { PopupConfirmation } from './PopupConfirmation/PopupConfirmation';
export { PopupDialog } from './PopupDialog/PopupDialog';
export { PopupInfo } from './PopupInfo/PopupInfo';
export { SearchPanel } from './SearchPanel/SearchPanel';
export { Spinner } from './Spinner/Spinner';
export { Tab } from './TabBar/Tab/Tab';
export { TabBar } from './TabBar/TabBar';
export { TextField } from './TextField/TextField';
