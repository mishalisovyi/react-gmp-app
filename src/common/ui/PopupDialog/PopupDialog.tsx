import React, { MouseEventHandler, useMemo } from 'react';

import { ModalWindow } from 'common/hoc';
import { ModalWindowActionButtonProps } from 'common/interfaces';

interface PopupDialogProps {
  children: React.ReactNode,
  title: string;
  onPrimaryButtonClick: MouseEventHandler<HTMLButtonElement>
  onSecondaryButtonClick: MouseEventHandler<HTMLButtonElement>
  onClose: MouseEventHandler<HTMLButtonElement>
}

export function PopupDialog({
  children,
  title,
  onPrimaryButtonClick,
  onSecondaryButtonClick,
  onClose,
}: PopupDialogProps) {
  const primaryButtonAttributes: ModalWindowActionButtonProps = useMemo(() => ({
    show: true,
    text: 'Submit',
    onClick: onPrimaryButtonClick,
  }), [onPrimaryButtonClick]);

  const secondaryButtonAttributes: ModalWindowActionButtonProps = useMemo(() => ({
    show: true,
    text: 'Reset',
    onClick: onSecondaryButtonClick,
  }), [onSecondaryButtonClick]);

  return (
    <ModalWindow
      title={title}
      primaryButton={primaryButtonAttributes}
      secondaryButton={secondaryButtonAttributes}
      onClose={onClose}
    >
      {children}
    </ModalWindow>
  );
}
