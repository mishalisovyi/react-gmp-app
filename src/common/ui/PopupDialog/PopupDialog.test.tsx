import React, { ReactNode, ReactPortal } from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { Icon } from 'common/constants';

import { PopupDialog } from './PopupDialog';

const handlePrimaryButtonClick = jest.fn();
const handleSecondaryButtonClick = jest.fn();
const handlePopupClosing = jest.fn();

beforeAll(() => {
  ReactDOM.createPortal = (node: ReactNode): ReactPortal => node as ReactPortal;
});

it('should render dialog popup', () => {
  const tree = renderer.create(
    <PopupDialog title="test title" onPrimaryButtonClick={handlePrimaryButtonClick} onSecondaryButtonClick={handleSecondaryButtonClick} onClose={handlePopupClosing}>
      <div>test</div>
    </PopupDialog>,
  );

  expect(tree.toJSON()).toMatchSnapshot();
});

it('should handle dialog popup primary button click', async () => {
  render(
    <PopupDialog title="test title" onPrimaryButtonClick={handlePrimaryButtonClick} onSecondaryButtonClick={handleSecondaryButtonClick} onClose={handlePopupClosing}>
      <div>test</div>
    </PopupDialog>,
  );

  await userEvent.click(screen.getByText('Submit'));

  expect(handlePrimaryButtonClick).toHaveBeenCalled();
});

it('should handle dialog popup secondary button click', async () => {
  render(
    <PopupDialog title="test title" onPrimaryButtonClick={handlePrimaryButtonClick} onSecondaryButtonClick={handleSecondaryButtonClick} onClose={handlePopupClosing}>
      <div>test</div>
    </PopupDialog>,
  );

  await userEvent.click(screen.getByText('Reset'));

  expect(handleSecondaryButtonClick).toHaveBeenCalled();
});

it('should handle dialog popup closing', async () => {
  render(
    <PopupDialog title="test title" onPrimaryButtonClick={handlePrimaryButtonClick} onSecondaryButtonClick={handleSecondaryButtonClick} onClose={handlePopupClosing}>
      <div>test</div>
    </PopupDialog>,
  );

  await userEvent.click(screen.getByText(Icon.MULTIPLICATION_X));

  expect(handlePopupClosing).toHaveBeenCalled();
});
