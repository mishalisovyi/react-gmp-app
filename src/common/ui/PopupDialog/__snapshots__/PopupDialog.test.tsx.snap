// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render dialog popup 1`] = `
<div>
  <div />
  <div>
    <div>
      <button
        className="button button--transparent"
        onClick={[MockFunction]}
        type="button"
      >
        âœ•
      </button>
    </div>
    <header>
      test title
    </header>
    <div>
      <div>
        test
      </div>
    </div>
    <div>
      <button
        className="button button--secondary"
        onClick={[MockFunction]}
        type="button"
      >
        Reset
      </button>
      <button
        className="button button--primary"
        onClick={[MockFunction]}
        type="button"
      >
        Submit
      </button>
    </div>
  </div>
</div>
`;
